builtins/builtins.go

```diff
@@ -31,6 +31,32 @@ func Rnd(args ...runtime.Object) runtime.Object {
 	return runtime.Float(rand.Float64())
 }
 
+func Int2Bit(args ...runtime.Object) runtime.Object {
+
+	if len(args) != 2 {
+		return runtime.Errorf("wrong number of arguments. got=%d, want=2", len(args))
+	}
+
+	number, ok := args[0].(runtime.Int)
+	if !ok {
+		return runtime.Errorf("%s arguments not supported", args[0].Type())
+	}
+	if number.Sign() == -1 {
+		return runtime.Errorf("%s invalue is less than zero", args[0].Type())
+	}
+
+	lengthArg, lenOk := args[1].(runtime.Int)
+	if !lenOk {
+		return runtime.Errorf("%s arguments not supported", args[1].Type())
+	}
+	length := int(lengthArg.Uint64());
+	if number.BitLen() > length {
+		return runtime.Errorf("%v value requires more than %d bits", number, length)
+	}
+
+	return &runtime.Bitstring{Value: number.Value(), Unit: runtime.Bit, Padding: 4}
```

Comment:
The hard-coded padding value strikes me odd. Shouldn't it rather be something like this?
```suggestion
	return &runtime.Bitstring{Value: number.Value(), Unit: runtime.Bit, Padding: length-number.BitLen()}
```

Reply:
bug, fixed